static/js/settings_org.js

```diff
@@ -600,123 +517,62 @@ function save_discard_widget_status_handler(subsection) {
-                }
-            }
-            const item = $(this).attr('data-value');
-            update(item);
-        });
-        $(`#${opts.container_id} .dropdown_list_reset_button`).click(function () {
-            update(null);
-        });
-    };
-
-    const value = () => {
-        let val = $(`#${opts.container_id} #${opts.value_id}`).data('value');
-        if (val === null) {
-            val = '';
-        }
-        return val;
-    };
+};
 
-    return {
-        setup,
-        render,
-        register_event_handlers,
-        value,
+exports.default_code_language_widget = null;
+exports.notifications_stream_widget = null;
+exports.signup_notifications_stream_widget = null;
+
+exports.init_dropdown_widgets = () => {
+    const streams = stream_data.get_streams_for_settings_page();
+    const notification_stream_options = {
+        data: streams.map(x => {
+            const item = {
+                name: x.name,
+                value: x.stream_id.toString(),
+            };
+            return item;
+        }),
+        subsection: 'notifications',
+        default_text: i18n.t("Disabled"),
+        render_text: (x) => {return `#${x}`;},
+        null_value: -1,
     };
+    exports.notifications_stream_widget = settings_list_widget(
+        Object.assign({setting_name: 'realm_notifications_stream_id'},
+                      notification_stream_options));
+    exports.signup_notifications_stream_widget = settings_list_widget(
+        Object.assign({setting_name: 'realm_signup_notifications_stream_id'},
+                      notification_stream_options));
+    exports.default_code_language_widget = settings_list_widget({
+        setting_name: 'realm_default_code_block_language',
+        data: Object.keys(pygments_data.langs).map(x => {
+            return {
+                name: x,
+                value: x,
+            };
+        }),
+        subsection: 'other-settings',
+        default_text: i18n.t("No language set"),
+    });
 };
 
-exports.default_code_language_widget = exports.DropdownListWidget({
-    setting_name: 'realm_default_code_block_language',
-    data: Object.keys(pygments_data.langs).map(x => {
-        return {
-            name: x,
-            value: x,
-        };
-    }),
-    subsection: 'other-settings',
-    default_text: i18n.t("No language set"),
-});
-
 exports.build_page = function () {
     meta.loaded = true;
 
     loading.make_indicator($('#admin_page_auth_methods_loading_indicator'));
 
     // Populate notifications stream modal
     if (page_params.is_admin) {
-        const streams = stream_data.get_streams_for_settings_page();
-        exports.populate_notifications_stream_dropdown(streams);
-        exports.populate_signup_notifications_stream_dropdown(streams);
+        exports.init_dropdown_widgets();
         exports.default_code_language_widget.setup();
+        exports.notifications_stream_widget.setup();
+        exports.signup_notifications_stream_widget.setup();
```

Comment:
Should we just call the `.setup()` methods inside `init_dropdown_widgets`?  

I have sorta the same question for the `.register_event_handlers` calls below.

Reply:
Yes. I'd just written the functions to match the initial setup. I've simplified the interface that `settings_list_widget` exposes.